docker/test_env/besu.go

```diff
@@ -0,0 +1,211 @@
+package test_env
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+
+	tc "github.com/testcontainers/testcontainers-go"
+	tcwait "github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/blockchain"
+	"github.com/smartcontractkit/chainlink-testing-framework/docker"
+	"github.com/smartcontractkit/chainlink-testing-framework/logging"
+	"github.com/smartcontractkit/chainlink-testing-framework/utils/testcontext"
+)
+
+const (
+	//TODO use Tate's mirror?
+	BESU_IMAGE_TAG = "23.10"
+)
+
+type Besu struct {
+	EnvComponent
+	ExternalHttpUrl      string
+	InternalHttpUrl      string
+	ExternalWsUrl        string
+	InternalWsUrl        string
+	InternalExecutionURL string
+	ExternalExecutionURL string
+	generatedDataHostDir string
+	chainConfg           *EthereumChainConfig
+	consensusLayer       ConsensusLayer
+	l                    zerolog.Logger
+	t                    *testing.T
+}
+
+func NewBesu(networks []string, chainConfg *EthereumChainConfig, generatedDataHostDir string, consensusLayer ConsensusLayer, opts ...EnvComponentOption) *Besu {
+	g := &Besu{
+		EnvComponent: EnvComponent{
+			ContainerName: fmt.Sprintf("%s-%s", "besu", uuid.NewString()[0:8]),
+			Networks:      networks,
+		},
+		chainConfg:           chainConfg,
+		generatedDataHostDir: generatedDataHostDir,
+		consensusLayer:       consensusLayer,
+		l:                    log.Logger,
```

Comment:
does it make sense to pass the logger as parameter?

Reply:
you tell me :-) what would be the use case? unless there's a strong one I'd rather not bloat the constructor too much. Or... I could add a functional option for passing logger?

Also maybe by default we should initialise the logger using `logging.GetTestLogger(t)` even if `t` is `nil` at that point?

Reply:
@AnieeG for now I've implemented the latter

Reply:
we need to pass `*testing.T` anyway :/